linkedlist/loop_detection.py

```diff
@@ -0,0 +1,57 @@
+"""
+Given a circular linked list, implement an algorithm that returns the node
+at the beginning of the loop.
+Circular linked list: A corrupt linked list in which a node's next pointer
+points to an earlier node, so as to make a loop in the linked list.
+
+For example
+input: 0 --> 1 --> 2 --> 3 --> 4 --> 2 ( The same 2 as earlier)
+output: 2
+
+Algorithm:
+1. Create two pointers, FastPointer and SlowPointer
+2. Move FastPointer at a rate of 2 steps, and SlowPointer at a rate of 1 step
+3. When they collide, move SlowPointer to head. Keep FastPointer
+4. Move SlowPointer and FastPointer at a rate of one step. Return collision point.
+"""
+class Node:
+    def __init__(self,x):
+        self.val = x
+        self.next = None
+
+def loop_detection(head):
```

Comment:
Please remove this file `loop_detection.py`.  
Similiar algorithms have already been done in `linkedlist/is_cyclic.py` and `linkedlist/first_cyclic_node.py`.